logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetOrCreateOneToOneConversationUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.conversation.Member
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+
+class GetOrCreateOneToOneConversationUseCase(
+    private val conversationRepository: ConversationRepository,
+) {
+
+    suspend operator fun invoke(otherUserId: UserId): CreateConversationResult {
+        return suspending {
+            conversationRepository.getOne2OneConversationDetailsByUserId(otherUserId).flatMap { conversation ->
+                if (conversation != null) {
+                    Either.Right(conversation.conversation.id)
+                } else {
+                    conversationRepository.createGroupConversation(members = listOf(Member(otherUserId))).map { it.id }
+                }
+            }
+        }.fold(
+            { failure -> CreateConversationResult.Failure(failure) },
+            { conversationId -> CreateConversationResult.Success(conversationId) }
+        )
+    }
+
+}
+
+sealed class CreateConversationResult {
+    data class Success(val conversationId: ConversationId) : CreateConversationResult()
```

Comment:
You only want the conversationId and not he whole conversation object?

Reply:
I think for now we can try to work with ConversationId, or do you see any case that we would need the whole object using it in AR ?

Reply:
Only need an ID for now I'm fine with that. Not hard to change.